### PR TITLE
spec: added chrome headless as default js runner

### PR DESCRIPTION
**Pull request by vitoravelino**
So far the feature tests with js enabled were running through
poltergeist driver. As we know, PhantomJS was discontinued and
that's bad. PhantomJS was far from perfect and we've had a lot
of flaky issues with it.

Even knowing that most of those flaky issues were our fault, it's
always good to improve our workflow whenever possible. Based on this we
are adding chrome headless as our default js runner for our feature
tests.

Unfortunately we had to keep poltergeist for one scenario that we
couldn't make chrome headless work properly. That scenario is when the
dev runs the feature tests inside of the container. For bare metal and
travis, chrome headless works like a charm.

Signed-off-by: Vítor Avelino <vavelino@suse.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,4 +34,11 @@ RUN zypper addrepo https://download.opensuse.org/repositories/devel:languages:go
            libtool m4 make makeinfo && \
     zypper clean -a
 
+# Add Google Chrome repo and install it
+RUN zypper  -n ar https://dl.google.com/linux/chrome/rpm/stable/x86_64 Chrome && \
```
Comment (mssola) [inline Dockerfile]: Merge with the above `RUN` block.

File: Gemfile [hunk 2]
```diff
@@ -100,10 +100,13 @@ end
 
 group :test do
   gem "capybara", "~> 2.14.3"
-  gem "codeclimate-test-reporter", group: :test, require: nil
+  gem "capybara-screenshot", "~> 1.0.0"
+  gem "chromedriver-helper"
+  gem "codeclimate-test-reporter", require: false
```
Comment (mssola) [inline Gemfile]: Why did you remove the `test` group here ?

File: spec/support/capybara.rb [hunk 3]
```diff
@@ -1,11 +1,76 @@
...
+      console = page.driver.browser.manage.logs.get(:browser)&.reject do |log|
+        log.message =~ JS_CONSOLE_FILTER
+      end
+
+      p console.map(&:message).join("\n") if console.present?
+    end
+  end
+end
+
+# HACK: when running tests inside of a container we should use poltergeist
```
Comment (mssola) [inline spec/support/capybara.rb]: Maybe this comment should go on the line where you set `JAVASCRIPT_DRIVER`.

File: Gemfile (hunk 2 above)
Comment (vitoravelino) [reply Gemfile]: Because it's already inside of the test group.

**Comment by vitoravelino** [comment]
> In the commit message you are talking about PhantomJS being discontinued, wasn't it Poltergeist? If not, then can you remove the lines where we download phantomjs in Travis?

:+1:
